L1Trigger/L1TMuonEndCap/src/PatternRecognition.cc

```diff
@@ -416,8 +403,16 @@ void PatternRecognition::process_single_zone(
           int bx1 = bool(bx_shifter & (1<<1));
           int bx0 = bool(bx_shifter & (1<<0));
 
-          if (bx2 == 0 && bx1 == 1) {  // is lifetime up? (note: drift_time is not being used)
+          // is lifetime up?
+          if (drift_time == 2 && bx2 == 0 && bx1 == 1) {
+            is_lifetime_up = true;
+          } else if (drift_time == 1 && bx1 == 0 && bx0 == 1) {
             is_lifetime_up = true;
+          } else {
+            // WARNING: It won't work if drift_time is not 1 or 2. The
```

Comment:
No assert statements.  So, according to this warning, if bxWindow_ == 1, the logic will fail.  Should we insert logic to catch this wherever "bxWindow_" is read in from the python configuration (or FW version), rather than inside the PatternRecognition step?